### DISCUSSION TITLE
Q. 외부에서 주입 받은 의존성을 사용 하는 클래스(jwt)가 도메인인 것이 맞나?

### DISCUSSION THREAD

**Discussion by jurlring** [woowacourse-teams/2022-gong-seek#45] [Q&A]
- 현재는 application

**Comment by Gyuchool** [comment]
엄밀히 따지면 외부 라이브러리에 영향을 받아 변경이 가해질수 있기 때문에 도메인과 분리해야 한다고 생각합니다. 현재 인터페이스로 분리 되어있으니 어느정도는 변경에 대비된 것 같아요🤔

**Comment by jurlring** [comment]
jwt가 저희 어플리케이션에서 정말 중요하고 필요한 객체라고 생각되어 도메인에 있어야한다고 생각은 듭니다. 

외부에서 주입 받은 의존성을 사용한다는 것에 의한 변경점은 2가지라고 생각됩니다.
1. 라이브러리 버전에 따른 변경
2. 다른 라이브러리를 사용함에 의한 변경

이를 통해 겪는 문제점은 jwt로 들자면 1번의 경우 JwtTokenProvider 클래스에서 사용되는 Jwts의 메서드가 변경되거나, 2번의 경우Jwts가 아닌 다른 의존성을 사용하게 된다는 것이라고 생각합니다.

가장 중요한 것은 변경점을 최소화 해야한다는 것인데, 1번 같은 경우 인터페이스로 분리한다고 해서 변경을 최소화할 방법이 무엇인지는 아직 모르겠네요.
2번과 같은 경우를 본다면 클래스명을 수정해야하므로 인터페이스로 둔 후, 구현체 분리를 통해 해결할 수 있을 것 같아요.
